airbyte-db/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -41,15 +41,19 @@
  */
 public class DefaultJdbcDatabase implements JdbcDatabase {
 
-  private final DataSource ds;
+  private final CloseableConnectionSupplier connectionSupplier;
 
-  public DefaultJdbcDatabase(final DataSource ds) {
-    this.ds = ds;
+  public DefaultJdbcDatabase(final DataSource dataSource) {
+    this(new DataSourceConnectionSupplier(dataSource));
+  }
+
+  public DefaultJdbcDatabase(final CloseableConnectionSupplier connectionSupplier) {
```

Comment:
we don't get a datasource with snowflake so we need to wrap if we want to reuse the DefaultJdbcDatabase.